sdks/python/apache_beam/options/pipeline_options_validator_test.py

```diff
@@ -111,69 +111,65 @@ def get_validator(temp_location, staging_location):
-            'staging_location': 'gs://foo/bar',
-            'errors': ['temp_location']
-        },
-        {
-            'temp_location': 'gs://foo',
-            'staging_location': 'gs://foo/bar',
-            'errors': ['temp_location']
-        },
-        {
-            'temp_location': 'gs://foo/',
-            'staging_location': 'gs://foo/bar',
-            'errors': []
-        },
-        {
-            'temp_location': 'gs://foo/bar',
-            'staging_location': 'gs://foo/bar',
-            'errors': []
-        },
-    ]
+    test_cases = [{
+        'temp_location': None, 'staging_location': 'gs://foo/bar', 'errors': []
+    },
+                  {
+                      'temp_location': None,
+                      'staging_location': None,
+                      'errors': ['staging_location', 'temp_location']
+                  },
+                  {
+                      'temp_location': 'gs://foo/bar',
+                      'staging_location': None,
+                      'errors': []
+                  },
+                  {
+                      'temp_location': 'gs://foo/bar',
+                      'staging_location': 'gs://ABC/bar',
+                      'errors': []
+                  },
+                  {
+                      'temp_location': 'gcs:/foo/bar',
+                      'staging_location': 'gs://foo/bar',
+                      'errors': []
+                  },
+                  {
+                      'temp_location': 'gs:/foo/bar',
+                      'staging_location': 'gs://foo/bar',
+                      'errors': []
+                  },
+                  {
+                      'temp_location': 'gs://ABC/bar',
+                      'staging_location': 'gs://foo/bar',
+                      'errors': []
+                  },
+                  {
+                      'temp_location': 'gs://ABC/bar',
+                      'staging_location': 'gs://BCD/bar',
+                      'errors': ['temp_location', 'staging_location']
+                  },
+                  {
+                      'temp_location': 'gs://foo',
+                      'staging_location': 'gs://foo/bar',
+                      'errors': []
+                  },
+                  {
+                      'temp_location': 'gs://foo/',
+                      'staging_location': 'gs://foo/bar',
+                      'errors': []
+                  },
+                  {
+                      'temp_location': 'gs://foo/bar',
+                      'staging_location': 'gs://foo/bar',
+                      'errors': []
+                  }]
 
     for case in test_cases:
       errors = get_validator(case['temp_location'],
                              case['staging_location']).validate()
       self.assertEqual(
-          self.check_errors_for_arguments(errors, case['errors']), [])
+          self.check_errors_for_arguments(errors, case['errors']), [], case)
```

Comment:
Printing the entire case here is okay, but providing each case a name (even if it's not incredibly descriptive) will help self-document what each individual case is supposed to do

Reply:
True enough, but I would rather get this in with a small improvement and leave the major refactoring to a future PR.